### ISSUE TITLE
Python class variables cannot use list comprehension

### ISSUE THREAD

**Issue by zjqzero** [python/cpython#114904]
# Bug report

### Bug description:

```python
class Test:
    HEADERS = ['date']
    index_map = {field: idx for idx, field in enumerate(HEADERS)}
    indexes = [index_map[key] for key in HEADERS]
# raise NameError: name 'index_map' is not defined

class Test:
    HEADERS = ['date']
    index_map = {field: idx for idx, field in enumerate(HEADERS)}
    indexes = []
    for key in HEADERS:
        indexes.append(index_map[key])
# not any error
```

Is this the intended design? thx.

### CPython versions tested on:

3.10, 3.12

### Operating systems tested on:

Linux, Windows

**Comment by Delengowski** [comment]
I would give this a strong read.

https://stackoverflow.com/a/13913933/9063378

tldr; This is indeed an error, and by design.

**Comment by serhiy-storchaka** [comment]
It is a duplicate of #47942.
